### PR TITLE
Allow a user to set the id of the DefaultLockRepository via ctor

### PR DESCRIPTION
**Pull request by cppwfs**
resolves INT-4353

### CODE REVIEW

**Comment by artembilan** [comment]
Thank you, @cppwfs , for the contribution!

Please, consider to rebase to the latest `master` to pick up the SF 5.0 GA upgrade to avoid the problem with "Bad Gateway" mentioned in Travis report.

**Comment by cppwfs** [comment]
@artembilan rebased.  Thanks for the heads up!

**Comment by artembilan** [comment]
Np. I have just noticed myself that my eyes are blind with all that SF goodies and, therefore, forgot to upgrade here 😄 

However according the consistence failure on CI for similar "Bad Gateway" I would say we are not guilty - something bad with our `/release` repo. Maybe too many people tries to upgrade their project to SF 5.0 GA at the same time 😉

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
 	 */
 	public static final int DEFAULT_TTL = 10000;
 
-	private final String id = UUID.randomUUID().toString();
+	private String id = UUID.randomUUID().toString();
```
Comment (artembilan) [inline spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java]: No, this still has to be `final` and the default value must be populated from the default ctor.

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java [hunk 2]
```diff
@@ -86,6 +86,19 @@ public DefaultLockRepository(DataSource dataSource) {
 	}
 
 	/**
+	 * Constructor that allows the user to specify a client id that will
+	 * be associated for all the locks persisted by the store.
+	 * @param dataSource the {@link DataSource} used to maintain the lock repository.
+	 * @param id the client id to be associated with locks handled by the repository.
```
Comment (artembilan) [inline spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java]: `@since 4.3.13`

**Comment by cppwfs** [comment]
I completely agree.  Everyone wanting to play with the new toys ;-).

**Review by artembilan — CHANGES_REQUESTED** [review]
1. Your name to the `@author` list of affected classes
2. Update Copyright 
3. Some note about new `id` argument in the `jdbc.adoc`

Otherwise LGTM!

Feel free to reject all my concerns and I'll do the work for you during merge!

**Comment by cppwfs** [comment]
Updated the PR based on your review.  Thanks for the review!!!

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java [hunk 3]
```diff
@@ -79,9 +80,30 @@
 
 	private String countQuery = "SELECT COUNT(REGION) FROM %SLOCK WHERE REGION=? AND LOCK_KEY=? AND CLIENT_ID=? AND CREATED_DATE>=?";
 
-
+	/**
+	 * Constructor that initializes the client id that will be associated for
+	 * all the locks persisted by the store instance to a random {@link UUID}.
+	 * @param dataSource the {@link DataSource}used to maintain the lock repository.
```
Comment (artembilan) [inline spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java]: Space before `used` word

File: spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java [hunk 4]
```diff
@@ -79,9 +80,30 @@
...
+	 */
 	@Autowired
 	public DefaultLockRepository(DataSource dataSource) {
+		Assert.notNull(dataSource, "dataSource must not be null");
+		this.template = new JdbcTemplate(dataSource);
+		this.id = UUID.randomUUID().toString();
```
Comment (artembilan) [inline spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java]: Nit-pick, but we can just delegate all the hard work to the new ctor from here

**Review by artembilan — APPROVED** [review]
That's all
Will fix on merge.
Tomorrow already...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/e515a0d6432b782193c97f0d2387934ab495d3cd and cherry-picked into `4.3.x` as https://github.com/spring-projects/spring-integration/commit/b82973660bfe722dbba6d7e0899d7a14c783fc7d after resolving conflicts.

Glenn, thank you for the contribution.
Looking forward for more! 😉

**Pull request closed without merging**